bench/Main.hs

```diff
@@ -0,0 +1,101 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes       #-}
+{-# LANGUAGE TemplateHaskell   #-}
+
+-- | Benchmarks for @fmt@ library.
+
+module Main where
+
+import           Control.DeepSeq         (NFData)
+import           Data.Monoid             ((<>))
+import           Data.String             (IsString)
+import           Data.String.Interpolate (i)
+import           Data.Text               (Text)
+import qualified Data.Text               as T
+import qualified Data.Text.Format        as TF
+import           Data.Text.Format.Params as TF
+import qualified Data.Text.Lazy          as LT
+import           Fmt                     (( #| ), (|#))
+import           Formatting              (formatToString, int, sformat, stext, string,
+                                          (%))
+import           Text.Printf             (printf)
+
+import           Criterion               (Benchmark, bench, bgroup, nf)
+import           Criterion.Main          (defaultMain)
+
+----------------------------------------------------------------------------
+-- Format utility functions
+----------------------------------------------------------------------------
+
+format' :: TF.Params ps => TF.Format -> ps -> Text
+format' f = LT.toStrict . TF.format f
+
+formatS :: TF.Params ps => TF.Format -> ps -> String
+formatS f = LT.unpack . TF.format f
+
+tshow :: Show a => a -> Text
+tshow x = T.pack (show x)
+
+----------------------------------------------------------------------------
+-- Benchmarks utility functions
+----------------------------------------------------------------------------
+
+bGenericStringGroup :: (NFData s, IsString s) => String -> a -> [(String, a -> s)] -> Benchmark
+bGenericStringGroup sTag benchObj = bgroup sTag . map (\(tag, fmt) -> bench tag (nf fmt benchObj))
+{-# INLINABLE bGenericStringGroup #-}
```

Comment:
`INLINABLE` doesn't mean “maybe inline it”, it means “include the definition of this function into interface file so that when it's imported by another module it would be theoretically possible to inline it”. Since it's the main module, `INLINABLE` is not needed here.